contracts/TreasureMarketplace.sol

```diff
@@ -84,256 +114,235 @@ contract TreasureMarketplace is OwnableUpgradeable, PausableUpgradeable, Reentra
         uint256 pricePerItem
     );
 
-    event ItemCanceled(address seller, address nftAddress, uint256 tokenId);
-
-    /// @dev check if NFT is approved
-    /// @param _nft address of the NFT
-    modifier onlyApprovedToken(address _nft) {
-        require(tokenApprovals[_nft] != TokenApprovalStatus.NOT_APPROVED, "token is not approved for trading");
-        _;
-    }
-
-    /// @dev initializer
-    /// @param _fee fee to be paid on each sale, in basis points
-    /// @param _feeRecipient wallet to collets fees
-    /// @param _paymentToken address of the token that is used for settlement
-    function initialize(uint256 _fee, address _feeRecipient, IERC20Upgradeable _paymentToken) external initializer {
+    /// @notice Perform initial contract setup
+    /// @dev    The initializer modifier ensures this is only called once, the owner should confirm this was properly
+    ///         performed before publishing this contract address.
+    /// @param  initialFee          fee to be paid on each sale, in basis points
+    /// @param  initialFeeRecipient wallet to collets fees
+    /// @param  initialPaymentToken address of the token that is used for settlement
+    function initialize(
+        uint256 initialFee,
+        address initialFeeRecipient,
+        IERC20Upgradeable initialPaymentToken
+    )
+        external initializer
+    {
         __Ownable_init_unchained();
         __Pausable_init_unchained();
         __ReentrancyGuard_init_unchained();
 
-        setFee(_fee);
-        setFeeRecipient(_feeRecipient);
-        paymentToken = _paymentToken;
+        setFee(initialFee);
+        setFeeRecipient(initialFeeRecipient);
+        paymentToken = initialPaymentToken;
     }
 
-    /// @dev Creates an NFT listing
-    /// @param _nftAddress address of the NFT to be sold
-    /// @param _tokenId token ID of the NFT to be sold
-    /// @param _quantity number of tokens to be sold, for ERC721 must be 1
-    /// @param _pricePerItem amount of payment token (ERC20) chrged for each sold token (ERC1155) or for 1 NFT (ERC721)
-    /// @param _expirationTime timestamp after which listing is invalid
+    /// @notice Creates an item listing. You must authorize this marketplace with your item's token contract to list.
+    /// @param  nftAddress     which token contract holds the offered token
+    /// @param  tokenId        the identifier for the offered token
+    /// @param  quantity       how many of this token identifier are offered (or 1 for a ERC-721 token)
+    /// @param  pricePerItem   the price (in units of the paymentToken) for each token offered
+    /// @param  expirationTime UNIX timestamp after when this listing expires
     function createListing(
-        address _nftAddress,
```

Comment:
It's not part of the style guide however I adapted a convention to prepend function vars with an underscore for an easier distinction of vars.